### PR TITLE
Config: check environment for Grafana admin URL, username, and password

### PR DESCRIPTION
**Pull request by jalevin**
**What this PR does**:
This PR adds a check for the GRAFANA_ADMIN_USER, and GRAFANA_ADMIN_PASSWORD environment variables before falling back to defaults following the [conventions for GRAFANA_VERSION](https://github.com/grafana/plugin-tools/blob/main/packages/plugin-e2e/src/fixtures/grafanaVersion.ts#L6).

For the baseURL we add a recommendation for the playwright config to use GRAFANA_URL or fallback to localhost:3000.

**Why we need it**:
There are circumstances for testing where we would like to run tests against an instance of Grafana that is not controlled by the test environment. For that we need to pass in the url, username, and password of the grafana instance into the test. Adding support for these environment variables allows us to do this without changing the configuration of the playwright config and makes the tests more portable without intervention from the author

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/create-plugin@5.7.0-canary.1244.aad8b0e.0
  npm install @grafana/plugin-e2e@1.10.0-canary.1244.aad8b0e.0
  # or 
  yarn add @grafana/create-plugin@5.7.0-canary.1244.aad8b0e.0
  yarn add @grafana/plugin-e2e@1.10.0-canary.1244.aad8b0e.0
  ```
</details>

### CODE REVIEW

**Review by sunker — COMMENTED** [review]
Thanks for raising this PR @jalevin!

Can we change the name of the env variables from `GRAFANA_USER` and `GRAFANA_PASSWORD` to `GRAFANA_ADMIN_USER` and `GRAFANA_ADMIN_PASSWORD`? 

Please note that the `packages/plugin-e2e/playwright.config.ts` file is not exposed to plugins - it's only used to test the internals of the plugin-e2e package. When new plugins are scaffolded using the [create-plugin tool](https://grafana.com/developers/plugin-tools/#scaffold-a-plugin), [this](https://github.com/grafana/plugin-tools/blob/main/packages/create-plugin/templates/common/playwright.config) template file is used to generate a playwright config. You may want to add the `GRAFANA_URL` variable there instead.

File: packages/plugin-e2e/playwright.config.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default defineConfig<PluginOptions>({
   /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
   use: {
     /* Base URL to use in actions like `await page.goto('/')`. */
-    baseURL: 'http://localhost:3000',
+    baseURL: process.env.GRAFANA_URL || 'http://localhost:3000',
```
[the remark this answers is not in the document]
Comment (jalevin) [reply packages/plugin-e2e/playwright.config.ts:24]: Ah! thanks. Will fix.

**Comment by jalevin** [comment]
> Thanks for raising this PR @jalevin!
> 
> Can we change the name of the env variables from `GRAFANA_USER` and `GRAFANA_PASSWORD` to `GRAFANA_ADMIN_USER` and `GRAFANA_ADMIN_PASSWORD`?
> 

You bet! I _think_ that will work. Let me coordinate some changes on my end and I'll circle back.

**Comment by jalevin** [comment]
@sunker We've updated Bench and all of our repos to follow the new convention for `GRAFANA_ADMIN_USER` and `GRAFANA_ADMIN_PASSWORD`. I think this will be more clear across the board for all of our repos. Thanks for the callout!

**Review by sunker — APPROVED** [review]
Nice! Please note that the changes in the playwright config will _only_ have an impact on new plugins that are scaffolded with the `create-plugin` tool from now on (using a version of create plugin where this change is included). The playwright config is entirely owned by the plugin, so it also won't have an impact on plugin that run the `create-plugin update` command. 

Feel free to merge when you want. If you'd like to do some testing before merging, you can install the pre-release. See PR description for pre-release version number.
Screenshot 2024-10-24 at 08 56 03

**Pull request merged**